tc-messaging/src/main/java/com/tc/entity/MessageCodecSupplier.java

```diff
@@ -0,0 +1,31 @@
+/*
+ *
+ *  The contents of this file are subject to the Terracotta Public License Version
+ *  2.0 (the "License"); You may not use this file except in compliance with the
+ *  License. You may obtain a copy of the License at
+ *
+ *  http://terracotta.org/legal/terracotta-public-license.
+ *
+ *  Software distributed under the License is distributed on an "AS IS" basis,
+ *  WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for
+ *  the specific language governing rights and limitations under the License.
+ *
+ *  The Covered Software is Terracotta Core.
+ *
+ *  The Initial Developer of the Covered Software is
+ *  Terracotta, Inc., a Software AG company
+ *
+ */
+package com.tc.entity;
+
+import com.tc.object.EntityID;
+import org.terracotta.entity.EntityMessage;
+import org.terracotta.entity.EntityResponse;
+import org.terracotta.entity.MessageCodec;
+
+/**
+ *
+ */
+public interface MessageCodecSupplier<M extends EntityMessage, R extends EntityResponse> {
```

Comment:
It looks like we aren't really able to capitalize on the generics whenever we are using this interface so it seems like we should use the EntityMessage and EntityResponse, directly, and rely on standard inheritance rules and down-casting.